### PR TITLE
Add Basic Rules Visualization

### PR DESCRIPTION
**Pull request by chikeen**
# Summary

- Added @material-ui/core and @material-ui/icon to package.json
- Added a simple table that display all the rules from IIGO, under IIGO tab
- * Use new changes from main-adding-to-gamestate (in case that hasn't been merged)

## Test Plan
- not applicable. Are we writing test for Visualisation?

Screenshot 2021-01-07 at 12 08 00
Screenshot 2021-01-07 at 12 08 09

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

**Comment by chikeen** [comment]
one sec, checking failed test case.

**Comment by lhl2617** [comment]
we're not testing vis yea :)

**Comment by lhl2617** [comment]
screenshots would be nice

File: website/src/components/Visualisations/IIGO/Util/ProcessedData.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+export type RuleType = {
+    ruleName: string;
+    mutable: boolean;
+    linked: boolean;
+    variables: string[];
+    history: {season: number; turn: number}[];
+};
+
+export const processRulesData = (data: OutputJSONType) :RuleType[]=> {
```
Comment (lhl2617) [inline website/src/components/Visualisations/IIGO/Util/ProcessedData.ts]:
```suggestion
export const processRulesData = (data: OutputJSONType): RuleType[]=> {
```

File: website/src/components/Visualisations/IIGO/Util/ProcessedData.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    if (data.GameStates.length === 0) return []
+
+    // return CurrentRulesInPlay keys in term of seasons
+    let rulesInSeasons = data.GameStates.map((episode) => {
```
Comment (lhl2617) [inline website/src/components/Visualisations/IIGO/Util/ProcessedData.ts]:
```suggestion
    const rulesInSeasons = data.GameStates.map((episode) => {
```

File: website/src/components/Visualisations/IIGO/Util/ProcessedData.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return {
+            season: episode.Season,
+            turn: episode.Turn,
+            rules: Object.keys(episode.CurrentRulesInPlay)
+        }
+    })
+
+    // return a list of rules of RuleType
+    let rulesDict:any = {}
+    // each season, do...
```
Comment (lhl2617) [inline website/src/components/Visualisations/IIGO/Util/ProcessedData.ts:25]: can probably change to `.map` then `.flat`, but ok

Comment (lhl2617) [reply website/src/components/Visualisations/IIGO/Util/ProcessedData.ts:25]: also probably don't need `any` here

File: website/src/components/Visualisations/IIGO/IIGO.tsx [hunk 4]
```diff
@@ -1,15 +1,126 @@
...
+import Typography from '@material-ui/core/Typography';
+import Paper from '@material-ui/core/Paper';
+import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
+import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';
+
+import List from '@material-ui/core/List';
+import ListItemText from '@material-ui/core/ListItemText';
+
+const useRowStyles = makeStyles({
+    root: {
```
Comment (lhl2617) [inline website/src/components/Visualisations/IIGO/IIGO.tsx]: can u change indentation to 2 spaces? i think the file was orig. 2 space

Comment (chikeen) [reply website/src/components/Visualisations/IIGO/IIGO.tsx]: hmm, I think the original files has 2 spaces indentation, but all the edited components switched to 4 space. Check Roles.
Therefore, I change all of mine to 4 spaces too.

Comment (lhl2617) [reply website/src/components/Visualisations/IIGO/IIGO.tsx]: ok, we're setting up prettier soon so thats fine

**Comment by lhl2617** [comment]
@chikeen need to run prettier

**Review by yp717 — DISMISSED** [review]
Logs show that this is failing linter changes - can you please update with latest version of main to get linter support @chikeen

File: website/src/components/Visualisations/IIGO/IIGO.tsx [hunk 5]
```diff
@@ -1,15 +1,124 @@
...
+                        ))}
+                    </TableBody>
+                </Table>
+            </TableContainer>
+        </div>
+  );
 }
 
-export default IIGO
+
```
Comment (yp717) [inline website/src/components/Visualisations/IIGO/IIGO.tsx]: You could remove these extra line spaces but feel free to ignore this comment is pedantic

**Comment by chikeen** [comment]
should work now. Please merge. This back and forth are tiring.

**Review by lhl2617 — APPROVED** [review]
ship now iterate later

**Pull request merged**